ocaml/typing/typecore.ml

```diff
@@ -956,14 +956,15 @@ let enter_variable
     end else
       Ident.create_local name.txt
   in
+  let pv_uid = Uid.mk ~current_unit:(Env.get_unit_name ()) in
```

Comment:
It is not enough to simply create a `Uid.t` and attach it to the typedtree, it also needs to end up in the environment as the `Uid.t` for this binding. That means that it needs to go into `tps_pattern_variables` as a new field of the `pattern_variable` type. Then it's value should be used instead of the calls to `Uid.mk` in `add_pattern_variables` and `type_class_arg_pattern`.

Reply:
Something like this? a39546d6593f19f23f4fe007870a72f228b54d21